app/api/events.py

```diff
@@ -321,6 +354,129 @@ def remove_participant(request: Request, id: str, uid: str, token: AccessTokenPa
+                    if p["penalty"] >= 2 and new_pos <= pen_start_pos:
+                        raise HTTPException(400, "User with penalty can't be rearranged")
+
+                    # no need to swap same index
+                    if new_pos == i:
+                        continue
+                    # swaps current pos with new pos
+                    participants[i], participants[new_pos] = participants[new_pos], participants[i]
+                    break
+
+        if event["maxParticipants"]:
+            for i, p in enumerate(participants):
+                # ensure fields exist
+                p = Participant.parse_obj(p).dict()
+                if p["confirmed"] and i >= event["maxParticipants"]:
+                   raise HTTPException(400, "Confirmed user cannot be moved to a non confirmed spot") 
+
+        res = db.events.find_one_and_update({'eid': event["eid"]}, {"$set": { "participants": participants}})
+
+        if res == None:
+            HTTPException(500, "Unexpected error when updating participants list")
+
+        return Response(status_code=200)
+
+@router.post('/{id}/confirm', dependencies=[Depends(validate_uuid)])
+async def confirmation(request: Request, id: str, token: AccessTokenPayload = Depends(authorize_admin)):
+    async with lock:
+        db = get_database(request)
+        event = get_event_or_404(db, id)
+        num_confs = num_of_confirmed_participants(event["participants"])
+
+        if event["bindingRegistration"] == False:
+            raise HTTPException(400, "Events without bindingRegistration should not send out confirmations")
+
+        if event_has_started(event):
+            raise HTTPException(400, "Cannot send confirmation after event start")
+
+        if event["public"] == False:
+            raise HTTPException(400, "Cannot send confirmation to a unpublished event")
+
+        # if users cannot join confirmations cannot be sent out
+        if not valid_registration(event["registrationOpeningDate"]):
+            raise HTTPException(400, "Cannot send confirmations before registration is opened")
+
+        result = db.events.find_one_and_update(
+            {'eid': UUID(id)},
+            {"$set": {"confirmed": True}})
+
+        if not result:
+            raise HTTPException(500, "Unexpected error when updating event")
+
+        # all users joined gets confirmed if maxParticipants is not set
+        confirmationPositions = len(event["participants"]) - num_confs
+        if event["maxParticipants"] != None:
+            confirmationPositions = event["maxParticipants"] - num_confs
+            if confirmationPositions == 0:
+                raise HTTPException(400, "All participants have received confirmation ")
+
+        # Aggregate all participants emails to send confirmation email
+        # doesn't needs to filter out penalty < 2 as these should be at the bottom of the list
+        # meaning $limit excludes penalized participants
+        pipeline = [
```

Comment:
does this keep order by default or do we need to be explicit about sort here or something? @Sagensagen

Reply:
I believe it comes out the way it's inserted, but I think it would be better to be explicit and order by index